### PR TITLE
Enable nilerr linter & fix findings

### PR DESCRIPTION
**Pull request by jtraglia**
Enable the [nilerr](https://github.com/gostaticanalysis/nilerr) linter in `golangci-lint` and fix a few of its findings, which I believe are minor but valid.

**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**

There are four calls to `ValidatorAtIndex` that will return `false, nil` instead of `false, err`. Like this:
https://github.com/prysmaticlabs/prysm/blob/ff1b03ab13bfbc487203aac8879f71fb66bd960e/beacon-chain/core/helpers/sync_committee.go#L37-L40

I'm not sure if this is error state is even reachable, but just in case it is, I think we should return `err`.

In other places, this is handled as expected. Like this:
https://github.com/prysmaticlabs/prysm/blob/ff1b03ab13bfbc487203aac8879f71fb66bd960e/beacon-chain/core/blocks/signature.go#L76-L79

But if returning `nil` there is intended, I think it's worthy of a comment.

----

Edit: this also fixes some findings from the CI lint check (I didn't see these locally).

For one's that seemed fine, I added: `//nolint:nilerr`

### CODE REVIEW

**Review by terencechain — DISMISSED** [review]
thanks!

**Comment by jtraglia** [comment]
Hmm the CI has more findings of `nilerr`. I didn't see those locally. Will check them out.

**Review by terencechain — APPROVED** [review]

**Review by terencechain — APPROVED** [review]

File: testing/endtoend/components/eth1/transactions.go [hunk 1]
```diff
@@ -110,11 +110,11 @@ func SendTransaction(client *rpc.Client, key *ecdsa.PrivateKey, f *filler.Filler
...
-				return nil
+				return err
 			}
 			signedTx, err := types.SignTx(tx, types.NewLondonSigner(chainid), key)
 			if err != nil {
-				return nil
+				return err
 			}
 			err = backend.SendTransaction(context.Background(), signedTx)
 			return nil
```
Comment (prestonvanloon) [inline testing/endtoend/components/eth1/transactions.go:120]:
```suggestion
			return backend.SendTransaction(context.Background(), signedTx)
```

Another nil err return here

**Review by terencechain — APPROVED** [review]

**Comment by prestonvanloon** [comment]
This is failing some unit tests. Possibly due to poor test setup.

**Review by rkapka — APPROVED** [review]

**Review by rkapka — APPROVED** [review]

**Pull request merged**